run/O2HitMerger.h

```diff
@@ -424,11 +473,10 @@ class O2HitMerger : public FairMQDevice
     // but iteration over keys of a standard map in C++ is ordered
 
     // b) merge the general data
-    merge<std::vector<o2::MCTrack>>("MCTrack", *tree, *mOutTree);
-    // TODO: fix track numbers in TrackRefs
-    merge<std::vector<o2::TrackReference>>("TrackRefs", *tree, *mOutTree);
-    merge<o2::dataformats::MCTruthContainer<o2::TrackReference>>("IndexedTrackRefs",
-                                                                 *tree, *mOutTree);
+    //
+    // for MCTrack remap the motherIds and merge at the samee go
+    remapTrackIdsAndMerge<std::vector<o2::MCTrack>>("MCTrack", *tree, *mOutTree, trackoffsets);
+    remapTrackIdsAndMerge<std::vector<o2::TrackReference>>("TrackRefs", *tree, *mOutTree, trackoffsets);
```

Comment:
we need to keep the `o2::dataformats::MCTruthContainer` for now to not fail the test. In the end we can decide which format to keep. This one here allows to easiky query TrackRefs given a trackID.